setup.sh

```diff
@@ -98,112 +84,28 @@ function _preparation() {
-                    apt-get install -q -y linux-image-686 >> "${log}" 2>&1
-                elif [[ $arch == x86_64 ]]; then
-                    apt-get install -q -y linux-image-amd64 >> "${log}" 2>&1
-                fi
-            fi
-            mv /etc/grub.d/06_OVHkernel /etc/grub.d/25_OVHkernel
-            update-grub >> "${log}" 2>&1
-        fi
-    fi
+    bash /etc/swizzin/scripts/nukeovh
 }
 
 function _intro() {
     whiptail --title "Swizzin seedbox installer" --msgbox "Yo, what's up? Let's install this swiz." 7 43
 }
 
 function _adduser() {
-    while [[ -z $user ]]; do
-        user=$(whiptail --inputbox "Enter username for Swizzin \"master\"" 8 40 3>&1 1>&2 2>&3)
-        exitstatus=$?
-        if [ "$exitstatus" = 1 ]; then exit 0; fi
-        if [[ $user =~ [A-Z] ]]; then
-            read -n 1 -s -r -p "Usernames must not contain capital letters. Press enter to try again."
-            printf "\n"
-            user=
-        elif [[ $user =~ ("swizzin"|"admin"|"root") ]]; then
-            read -n 1 -s -r -p "$user is a reserved username -- please use something else. Press enter to try again."
-            printf "\n"
-            user=
-        fi
-    done
-    while [[ -z "${pass}" ]]; do
-        pass=$(whiptail --inputbox "Enter new password for $user. Leave empty to generate." 9 40 3>&1 1>&2 2>&3)
-        exitstatus=$?
-        if [ "$exitstatus" = 1 ]; then exit 0; fi
-        if [[ -z "${pass}" ]]; then
-            pass="$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c16)"
-        fi
-        if [[ -n $(which cracklib-check) ]]; then
-            echo "Cracklib detected. Checking password strength."
-            sleep 1
-            str="$(cracklib-check <<< "$pass")"
-            check=$(grep OK <<< "$str")
-            if [[ -z $check ]]; then
-                read -n 1 -s -r -p "Password did not pass cracklib check. Press any key to enter a new password"
-                printf "\n"
-                pass=
-            else
-                echo "OK."
-            fi
-        fi
-    done
+    #shellcheck source=sources/functions/utils
+    . /etc/swizzin/sources/functions/utils
+    username_check whiptail
+    password_check whiptail
     echo "$user:$pass" > /root/.master.info
-    if [[ -d /home/"$user" ]]; then
-        echo "User directory already exists ... "
-
-        echo "Changing password to new password"
-        chpasswd <<< "${user}:${pass}"
-        htpasswd -b -c /etc/htpasswd $user $pass
-        mkdir -p /etc/htpasswd.d/
-        htpasswd -b -c /etc/htpasswd.d/htpasswd.${user} $user $pass
-        chown -R $user:$user /home/${user}
+    bash /etc/swizzin/scripts/box adduser "$user" "$pass" # TODO make it so that the password does not hit the logs
+    #TODO this should match word exactly, because amking user test, cancaelling, and making test1 will make no sudo modifications
+    if grep ${user} /etc/sudoers.d/swizzin > /dev/null 2>&1; then
+        echo_warn "No sudoers modification made"
```

Comment:
Maybe instead of issuing a warning with this language we should say "No sudoers modifications requuired". Otherwise, people might assume it's a bug if there's a warning attached to it.

Reply:
I wonder if the user even needs to know about this
```suggestion
        echo_log_only "No sudoers modification made"
```

Reply:
Yeah, fair enough -- probably don't need to know